### PR TITLE
feat(subtypes): support subtypes for charts in the UI

### PR DESCRIPTION
**Pull request by gabe-lyons**
Supports subtypes in UI for charts on search, entity page & lineage

## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: datahub-web-react/src/app/entity/chart/ChartEntity.tsx [hunk 1]
```diff
@@ -154,10 +154,12 @@ export class ChartEntity implements Entity<Chart> {
     getOverridePropertiesFromEntity = (chart?: Chart | null): GenericEntityProperties => {
         // TODO: Get rid of this once we have correctly formed platform coming back.
         const name = chart?.properties?.name;
+        const subTypes = chart?.subTypes;
         const externalUrl = chart?.properties?.externalUrl;
         return {
             name,
             externalUrl,
+            entityTypeOverride: subTypes ? capitalizeFirstLetterOnly(subTypes.typeNames?.[0]) : '',
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/entity/chart/ChartEntity.tsx:162]: should this be || Chart?

File: datahub-web-react/src/app/entity/chart/ChartEntity.tsx [hunk 2]
```diff
@@ -222,6 +225,7 @@ export class ChartEntity implements Entity<Chart> {
             type: EntityType.Chart,
             icon: entity?.platform?.properties?.logoUrl || undefined,
             platform: entity?.platform,
+            subtype: entity?.subTypes?.typeNames?.[0] || undefined,
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/entity/chart/ChartEntity.tsx:228]: bit weird that the "t' in this is lower cased

Comment (jjoyce0510) [reply datahub-web-react/src/app/entity/chart/ChartEntity.tsx:228]: is that intentional?

File: datahub-web-react/src/app/entity/chart/preview/ChartPreview.tsx [hunk 3]
```diff
@@ -76,7 +79,7 @@ export const ChartPreview = ({
             name={name || ''}
             urn={urn}
             description={description || ''}
-            type="Chart"
+            type={capitalizeFirstLetterOnly(subType) || 'Chart'}
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/entity/chart/preview/ChartPreview.tsx:82]: does capitalizeFirstLetterOnly handle case of undefined | null/

im assuming yes

**Review by jjoyce0510 — APPROVED** [review]
Small comments please check first

File: datahub-web-react/src/app/entity/chart/ChartEntity.tsx (hunk 1 above)
Comment (gabe-lyons) [reply datahub-web-react/src/app/entity/chart/ChartEntity.tsx:162]: no. null entity type override is safe. see https://github.com/datahub-project/datahub/blob/master/datahub-web-react/src/app/entity/dataset/DatasetEntity.tsx#L247

Comment (gabe-lyons) [reply datahub-web-react/src/app/entity/chart/ChartEntity.tsx:162]:
```
 getOverridePropertiesFromEntity = (dataset?: Dataset | null): GenericEntityProperties => {
        // if dataset has subTypes filled out, pick the most specific subtype and return it
        const subTypes = dataset?.subTypes;
        const extendedProperties: DatasetProperties | undefined | null = dataset?.properties && {
            ...dataset?.properties,
            qualifiedName: dataset?.properties?.qualifiedName || this.displayName(dataset),
        };
        return {
            name: dataset && this.displayName(dataset),
            externalUrl: dataset?.properties?.externalUrl,
            entityTypeOverride: subTypes ? capitalizeFirstLetterOnly(subTypes.typeNames?.[0]) : '',
            properties: extendedProperties,
        };
    };
```

File: datahub-web-react/src/app/entity/chart/ChartEntity.tsx (hunk 2 above)
Comment (gabe-lyons) [reply datahub-web-react/src/app/entity/chart/ChartEntity.tsx:228]: this is existing code. its intentional

File: datahub-web-react/src/app/entity/chart/preview/ChartPreview.tsx (hunk 3 above)
Comment (gabe-lyons) [reply datahub-web-react/src/app/entity/chart/preview/ChartPreview.tsx:82]: yes 🖖 

https://github.com/datahub-project/datahub/blob/399e032dfa2b4bf87b7b406e7b009e34e99a1003/datahub-web-react/src/app/shared/textUtil.ts#L17

File: datahub-web-react/src/app/entity/chart/ChartEntity.tsx (hunk 2 above)
Comment (gabe-lyons) [reply datahub-web-react/src/app/entity/chart/ChartEntity.tsx:228]: https://github.com/datahub-project/datahub/blob/399e032dfa2b4bf87b7b406e7b009e34e99a1003/datahub-web-react/src/app/entity/dataset/DatasetEntity.tsx#L329

**Pull request merged**